### PR TITLE
Started P2P rework

### PR DESCRIPTION
**Pull request by yueh**
Changed frequency to short.
Added a per grid RNG to request a new frequency, hopefully without many collisons.
Added a helper to convert between a frequency and 4 colours.

### CODE REVIEW

File: src/main/java/appeng/parts/p2p/PartP2PTunnel.java [hunk 1]
```diff
@@ -308,17 +308,17 @@ public boolean onPartShiftActivate( final EntityPlayer player, final EnumHand ha
 			final IMemoryCard mc = (IMemoryCard) is.getItem();
```
Comment (orod-org) [inline src/main/java/appeng/parts/p2p/PartP2PTunnel.java]: MAJOR Method appeng.parts.p2p.PartP2PTunnel.onPartShiftActivate(EntityPlayer, EnumHand, Vec3d) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)

**Comment by orod-org** [comment]
SonarQube analysis reported 9 issues
* MAJOR 5 major
* MINOR 2 minor
* INFO 2 info

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [PartP2PTunnel.java#L171](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d7cb7da7d560d838683f67bc23f1a73897a76edf/src/main/java/appeng/parts/p2p/PartP2PTunnel.java#L171): Method appeng.parts.p2p.PartP2PTunnel.onPartActivate(EntityPlayer, EnumHand, Vec3d) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [PartP2PTunnel.java#L181](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d7cb7da7d560d838683f67bc23f1a73897a76edf/src/main/java/appeng/parts/p2p/PartP2PTunnel.java#L181): Method appeng.parts.p2p.PartP2PTunnel.onPartActivate(EntityPlayer, EnumHand, Vec3d) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [Platform.java#L198](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d7cb7da7d560d838683f67bc23f1a73897a76edf/src/main/java/appeng/util/Platform.java#L198): Method appeng.util.Platform.formatPowerLong(long, boolean) creates array using constants [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3ASACM_STATIC_ARRAY_CREATED_IN_METHOD)
1. MAJOR [Platform.java#L817](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d7cb7da7d560d838683f67bc23f1a73897a76edf/src/main/java/appeng/util/Platform.java#L817): Method appeng.util.Platform.isWrench(EntityPlayer, ItemStack, BlockPos) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MINOR [StorageHelper.java#L249](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d7cb7da7d560d838683f67bc23f1a73897a76edf/src/main/java/appeng/spatial/StorageHelper.java#L249): Remove this use of "neighborChanged"; it is deprecated. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. INFO [Platform.java#L468](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d7cb7da7d560d838683f67bc23f1a73897a76edf/src/main/java/appeng/util/Platform.java#L468): Redundant nullcheck of out, which is known to be non-null in appeng.util.Platform.getBlockDrops(World, BlockPos) [rule](https://sonar.orod.org/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE)
1. INFO [Platform.java#L760](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d7cb7da7d560d838683f67bc23f1a73897a76edf/src/main/java/appeng/util/Platform.java#L760): Redundant nullcheck of n, which is known to be non-null in appeng.util.Platform.getItemDisplayName(Object) [rule](https://sonar.orod.org/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE)

**Comment by orod-org** [comment]
SonarQube analysis reported 9 issues
* MAJOR 5 major
* MINOR 2 minor
* INFO 2 info

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [PartP2PTunnel.java#L171](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/a6efd2351312bcf92201a721ccc6b39b1fcc3f2f/src/main/java/appeng/parts/p2p/PartP2PTunnel.java#L171): Method appeng.parts.p2p.PartP2PTunnel.onPartActivate(EntityPlayer, EnumHand, Vec3d) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [PartP2PTunnel.java#L181](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/a6efd2351312bcf92201a721ccc6b39b1fcc3f2f/src/main/java/appeng/parts/p2p/PartP2PTunnel.java#L181): Method appeng.parts.p2p.PartP2PTunnel.onPartActivate(EntityPlayer, EnumHand, Vec3d) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [Platform.java#L198](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/a6efd2351312bcf92201a721ccc6b39b1fcc3f2f/src/main/java/appeng/util/Platform.java#L198): Method appeng.util.Platform.formatPowerLong(long, boolean) creates array using constants [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3ASACM_STATIC_ARRAY_CREATED_IN_METHOD)
1. MAJOR [Platform.java#L817](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/a6efd2351312bcf92201a721ccc6b39b1fcc3f2f/src/main/java/appeng/util/Platform.java#L817): Method appeng.util.Platform.isWrench(EntityPlayer, ItemStack, BlockPos) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MINOR [StorageHelper.java#L249](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/a6efd2351312bcf92201a721ccc6b39b1fcc3f2f/src/main/java/appeng/spatial/StorageHelper.java#L249): Remove this use of "neighborChanged"; it is deprecated. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. INFO [Platform.java#L468](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/a6efd2351312bcf92201a721ccc6b39b1fcc3f2f/src/main/java/appeng/util/Platform.java#L468): Redundant nullcheck of out, which is known to be non-null in appeng.util.Platform.getBlockDrops(World, BlockPos) [rule](https://sonar.orod.org/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE)
1. INFO [Platform.java#L760](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/a6efd2351312bcf92201a721ccc6b39b1fcc3f2f/src/main/java/appeng/util/Platform.java#L760): Redundant nullcheck of n, which is known to be non-null in appeng.util.Platform.getItemDisplayName(Object) [rule](https://sonar.orod.org/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE)

**Comment by mgdgfbhevdyeh** [comment]
hello guys i have a problem
why disabled RF p2p now?

**Comment by mgdgfbhevdyeh** [comment]
hello guys i have a problem
why disabled RF p2p now?

**Comment by mindforger** [comment]
@mgdgfbhevdyeh i removed your double post, this is not an issue!
It is a pullrequest, a branch of code to be changed in future.
RF has been removed as the integration was not given anymore beyond 1.9, if you think this is an issue, open a new one, but do not answer here please :)

**Pull request closed without merging**